wmfdata/conda.py

```diff
@@ -0,0 +1,118 @@
+import json
+import os
+
+# In case conda is not installed, we still want these functions to return something useful.
+try:
+    from conda.cli.python_api import run_command as condacli
+    conda_installed = True
+except ImportError:
+    conda_installed = False
+
+from wmfdata.utils import log
+
+def info():
+    """
+    Returns the result of conda info --all --json as a dict.
+    """
+    if conda_installed:
+        return json.loads(condacli('info', ["--all", "--json"])[0])
+    else:
+        return {}
+
+def env_vars():
+    """
+    Returns dict of conda environment variables.
+    """
+    return info().get('env_vars', {})
+
+def active_prefix():
+    """
+    Returns the path to the active conda env, or None.
+    """
+    return info().get('active_prefix', None)
+
+def active_name():
+    """
+    Returns the name of the active conda env, or None.
+    """
+    return env_vars().get('CONDA_DEFAULT_ENV', None)
+
+def base_prefix():
+    """
+    If the active conda env is stacked on another conda env, returns
+    the path to that base conda env, or None.
+    """
+    return env_vars().get('CONDA_PREFIX_1', None)
+
+def is_active():
+    """
+    Returns True if a conda env is active, else False.
+    """
+    if active_prefix() is None:
+        return False
+    else:
+        return True
+
+def is_stacked():
+    """
+    Returns True if a conda env active and stacked on a base conda env, else False.
+    """
+    if base_prefix() is None:
+        return False
+    else:
+        return True
+
+def conda_base_env_path():
+    """
+    Returns the path to the conda base env, which on WMF servers is /usr/lib/anaconda-wmf.
+    This can be overridden by setting the CONDA_BASE_ENV_PATH env var.
+    """
+    return os.environ.get('CONDA_BASE_ENV_PATH', '/usr/lib/anaconda-wmf')
+
+conda_pack_defaults = {
+    # This is required to pack stacked conda envs.
+    'ignore_editable_packages': True,
+    'verbose': True,
+    'force': False,
+    'format': 'tgz',
+}
+
+def pack(**conda_pack_kwargs):
+    """
+    Calls conda_pack.pack.
+    If the packed output file already exists, this will not repackage
+    it unless conda_pack_kwargs['force'] == True.
+
+    Returns the path to the output packed env file.
+
+    Arguments:
+    * `conda_pack_kwargs` args to pass to conda_pack.pack().
+    """
+    kwargs = conda_pack_defaults.copy()
+    kwargs.update(conda_pack_kwargs)
+
+    # Make sure output is set to something, so we can return it if it already exists.
+    if 'output' not in kwargs:
+        conda_env_name = 'conda-env'
+        if 'prefix' in kwargs:
+            conda_env_name = os.path.basename(kwargs['prefix'])
+        elif 'name' in kwargs:
+            conda_env_name = kwargs['name']
+        elif is_active():
```

Comment:
what happens if there isn't an active conda env? will there be a `conda-conda-env.tgz` file or do we expect this to fail?

Reply:
Oo thanks for noticing the conda-conda thing.  Fixed.  but yes, the default output should be conda-env.tgz.  If no environment is active and `conda_pack_kwargs` aren't enough for `conda_pack.pack` to figure out what to pack, this will error.  I'll add a comment.